toString.js

```diff
@@ -0,0 +1,42 @@
+const LINES = {
+  r: '└──', //right
+  t: '├──', //T shape
+  v: '│  '  //vertical
+}
+
+function toString() {
+  console.log('Fastify Instance toString')
+  /* Print hooks */
+  console.log(`\nHooks`)
+  Object.keys(this._hooks).forEach((lfm, i) => {
+    const hooks = this._hooks[lfm]
+    console.log(`${
+      i < 3 ? LINES.t : LINES.r
+    } ${lfm} (${hooks.length} hooks): ${hooks.length > 0 ? hooks.map(h=>h.name) : '[]'}`)
+  })
+  /* Ideally: iterate over routes and print plugins & middleware by scope */
+  console.log(`\nRoutes`)
+  console.log(`${this.printRoutes()}`)
+
+  console.log(`\nMiddleware`)
```

Comment:
This should be "top level middlewares". I don't think you can see middlewares that are wrapped within a plugin.